### ISSUE TITLE
Attaching domain to package

### ISSUE THREAD

**Issue by sasa-mladenovic-troxo** [atomia/AtomiaStore#4]
Hi,

when I make an order with one domain and a package, and I connected that domain on step two to a package - I don't see the difference in the placed order in admin panel compared to the order where domain is not connected to package.

I thought I should see something like: _"Domain name: saletroxo12987309123.com"_ under package order name.

Maybe the problem is in `Atomia.Store.PublicOrderHandlers.CartItemHandlers.DefaultHandler.AmendOrder()`

                Add(order, new PublicOrderItem
                {
                    ItemId = Guid.Empty,
                    ItemNumber = item.ArticleNumber,
                    RenewalPeriodId = item.RenewalPeriodId,
                    Quantity = item.Quantity
                });

Perhaps this code should also set `PublicOrderItem.CustomData` based on `item.CartItem.CustomAttributes`?

**Comment by ivankustera** [comment]
Hi Sasa,

I'll try to help since Daniel is not here but have in mind I didn't work with Atomia Store a lot. However, this should match logic we have in Billing API and that exists in old Order Page too.

When you order one or more domains one of them should be main domain (connected to the package in this case). This makes difference what type of website is provisioned for this domain. This is determined by AtomiaService custom attribute on domain order line. If you connect domain to a package, it will have value CsLinuxWebsite which provisiones as apache website. If domain is not connected, it have default value which is CsDomainParking and is provisioned as parking for a domain. CsDomainParking is added for all additional domains on a order too. If selected package doesn't support adding websites (like DNS package), CsDomainParking is provisioned for main domain too.

So, long story short. If you connect domain to a package it is treated as a main package and gets AtomiaService = CsLinuxWebsite if package supports it. In all other cases domain will have AtomiaService = CsDomainParking. 

This is all added to domain order line. Package order line doesn't have any attributes to tell which domain is connected to it.

Ivan

**Comment by sasa-mladenovic-troxo** [comment]
Ok,

I have a request to allow ordering of more than one hosting packages in one order.
I understand that is possible in Atomia. THere is a multipackage flag that Slobodan told me about and I have set that to `true`. 

Is it possible to connect one domain to each ordered package? So that domain becomes default for that particular package - and so that domain get the CsLinuxWebsite if specific package allows that?

**Comment by ivankustera** [comment]
Hi,

In case of more than one package, you should add PackageGroupId custom attributes for all lines. Packages should have unique value for this attribute (I suggest sequential integers for value) and services should have the same value as packages they belong to. This way, they will be provisioned together with their package.

For example, if you order two packages they should have PackageGroupId 1 and 2. All domains that belong to the first package should have PackageGroupId = 1 and all domains for second package should have PackageGroupId = 2.

Ivan

**Comment by sasa-mladenovic-troxo** [comment]
Hi,

I used domain name as a PackageGroupId and it worked great. I also had to update Billing API configuration to allow multipackages with:

    <multiPackage enabled="True" />

@ivankustera thanks for the info ;)

I think we can mark this one as closed.
